src/aspire/source/simulation.py

```diff
@@ -70,11 +70,16 @@ def __init__(
 
         self.C = self.vols.n_vols
 
-        states = states or randi(self.C, n, seed=seed)
-        angles = angles or uniform_random_angles(n, seed=seed, dtype=self.dtype)
-
+        if states is None:
```

Comment:
We're protecting ourselves against `states` evaluating to `False` even if it's not `None`?

Reply:
I ran into trouble with the `unique_filters` again. I figured I'd just as soon have them (angles, unique_filters, states) be consistent. If you prefer the one liner I can restore.

Reply:
No this is much clearer. Was just wondering.